spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -1062,6 +1063,8 @@ class DistriOptimizer[T: ClassTag] (
 
     // unpersist the model because the next time optimize is called, new `models` will be
     // created
+    shutdown()
+    CachedModels.deleteKey(modelBroadcast.uuid)
```

Comment:
should this be a part of shutdonw()?

Reply:
Yes. Moving to `shutdown` makes it simple.